### DISCUSSION TITLE
Web - React - Custom wallet

### DISCUSSION THREAD

**Discussion by ethanpx** [WalletConnect/walletconnect-monorepo#3422] [Developer Support]
Hi Team.

I want to use the InjectedConnector from the library @wagmi/connectors/injected to integrate my custom wallet into Web3Modal through EthereumClient, but it's not working. The issue seems to be that the id in InjectedConnector is fixed, so my wallet is not being created. Is there any way to solve this problem? Thank you.

Here's my config:
![Screenshot 2023-08-10 at 13 38 49](https://github.com/WalletConnect/walletconnect-monorepo/assets/81727176/54385d00-695e-42dc-af0b-e6c954cef321)

**Comment by glitch-txs** [comment]
Hi, could you share a minimal implementation repo?
